core/sandbox/proxy/controller.go

```diff
@@ -119,9 +120,31 @@ func (s *remoteSandboxController) Shutdown(ctx context.Context, sandboxID string
 }
 
 func (s *remoteSandboxController) Wait(ctx context.Context, sandboxID string) (sandbox.ExitStatus, error) {
-	resp, err := s.client.Wait(ctx, &api.ControllerWaitRequest{SandboxID: sandboxID})
-	if err != nil {
-		return sandbox.ExitStatus{}, errdefs.FromGRPC(err)
+	// For remote sandbox controllers, the controller process may restart,
+	// we have to retry if the error indicates that it is the grpc disconnection.
+	var (
+		resp          *api.ControllerWaitResponse
+		err           error
+		retryInterval time.Duration = 128
+	)
+	for {
+		resp, err = s.client.Wait(ctx, &api.ControllerWaitRequest{SandboxID: sandboxID})
```

Comment:
Would you please add failpoint testcase to check if it's infinite loop for ttrpc shim? thanks

Reply:
I think maybe we need to add a mocked remote sandbox controller and then we can add failpoint testcast for this.

Reply:
Let's add that test in the follow-up.